kratos/tests/test_vector_interface.py

```diff
@@ -37,6 +37,30 @@ def test_list_construction(self):
         self.assertEqual(a[1],2)
         self.assertEqual(a[2],3)
 
+    def test_VectorSlice_construction(self):
+        v = Vector(5)
+        for i in range(len(v)):
+            v[i] = i
+
+        vec_slice = v[1:4]
+        v1 = Vector(vec_slice) # simple slice get

```

Comment:
can you please add a check with [isinstance](https://www.geeksforgeeks.org/python-unittest-assertisinstance-function/) to make sure this is a `VectorSlice`?

=> same for the other occurrences below